staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -43,10 +45,14 @@ type CustomResourceDefinitionVersion struct {
 	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
 	Name string
 	// Served is a flag enabling/disabling this version from being served via REST APIs
-	Served bool
-	// Storage flags the release as a storage version. There can be only one version
-	// flagged as Storage.
-	Storage bool
+	// Default is True.
+	// +optional
+	Served *bool
```

Comment:
See comment above. `DisableServing bool` would be much better than having this options. Actually, I think `+optional` is semantically wrong.

Reply:
In the API:

```
Versions:
- Name: v1
  Serve: False
- Name: v2
```

is cleaner than:
```
Versions:
- Name: v1
  DisableServing: True
- Name: v2
```

also I think we still have disagreement on having default value for non-optional fields. If a field does not have +optional, it will be marked as required in our spec and user need to provide it.